### ISSUE TITLE
Is it valid for wifi.sta.getapinfo() to return a null BSSID?

### ISSUE THREAD

**Issue by marcelstoer** [nodemcu/nodemcu-firmware#2025]
Running the example from https://nodemcu.readthedocs.io/en/latest/en/modules/wifi/#wifistagetapinfo I see that I get `nil` for my AP's BSSID. Is that an ok condition or does that signify a bug in the SDK or our mapping to Lua?

Maybe something for @dnc40085?

**Comment by dnc40085** [comment]
Yes, that is an ok condition.
From the [wifi.sta.getapinfo()](https://nodemcu.readthedocs.io/en/latest/en/modules/wifi/#returns_19) documentation
>`bssid` MAC address of Access Point, `nil` if no MAC address was configured

Unless your station was configured with the AP's mac address( ex: `wifi.sta.config({ssid=ssid, pwd=pwd, bssid="AA:BB:CC:DD:EE:FF"})`), the `bssid` field should be `nil`.

The reasoning behind this is that when no BSSID is set during station configuration, the `station_config.bssid_set` byte is set to `0` and the 6 byte `station_config.bssid` array is ignored and the corresponding location in flash is not modified. This location being unmodified could potentially contain a BSSID that does not correspond with the SSID so it must be ignored when the access point info is retrieved from flash.

Considering the confusion, maybe it would be prudent to change it to return an invalid mac address(`00:00:00:00:00:00`) instead of `nil`.

**Comment by marcelstoer** [comment]
Thanks for the explanation.
Before I posted this question I checked the documentation but "`nil` if no MAC address was configured" confused me. Why would I configure the AP's MAC I thought, it's got a MAC automatically. It didn't occur to me that this would be the MAC I set explicitly while configuring the station.
Then I looked at https://github.com/nodemcu/nodemcu-firmware/blob/dev/app/modules/wifi.c#L564 and tracked the use of `bssid_set`. However, not being a firmware developer this didn't fully resolve my confusion. Hence, this issue.

Conclusion: there's no way we could figure out the AP's MAC from Lua land unless we knew it beforehand?
